AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginTests/Sync/Support/ModelCompareTests.swift

```diff
@@ -0,0 +1,317 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import SQLite
+import XCTest
+
+@testable import Amplify
+@testable import AmplifyTestCommon
+@testable import AWSDataStoreCategoryPlugin
+@testable import AWSPluginsCore
+
+// swiftlint:disable type_body_length
+class ModelCompareTests: BaseDataStoreTests {
```

Comment:
can you add test for when a Post has a `nil` field value and the other Post has a value? they are not equal, right?